streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -988,6 +1005,14 @@ private void prepareChangelogs(final Map<TaskId, Task> tasks,
                 } catch (final Exception e) {
                     throw new StreamsException("State restore listener failed on batch restored", e);
                 }
+
+                final TaskId taskId = changelogMetadata.stateManager.taskId();
+                final Task task = tasks.get(taskId);
+                // if the log is truncated between when we get the log end offset and when we get the
+                // consumer position, then it's possible that the difference become negative and there's actually
+                // no records to restore; in this case we just initialize the sensor to zero
```

Comment:
Not sure if I can follow? If we get the end-offset first, the end-offset we have won't change even if the log gets truncated? And the consumer position should not change anyway? Can you elaborate?

Reply:
I was thinking about the case where the consumer's position is set as the checkpointed offset, whereas the log end offset was truncated after the checkpoint was written. Maybe I did not make that clear in the above comment? Will reword a bit.